### PR TITLE
v1.12 backports 2023-02-18

### PR DESCRIPTION
**Pull request by sayboras**
- [x] #23763 -- envoy: Avoid empty typeURL for all resources (@sayboras)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 23763; do contrib/backporting/set-labels.py $pr done 1.12; done
```

### CODE REVIEW

**Comment by sayboras** [comment]
/test-backport-1.12

**Comment by sayboras** [comment]
/test-1.17-4.9

**Comment by sayboras** [comment]
/test-backport-1.12

**Review by pchaigno — COMMENTED** [review]
Were there any conflicts?

**Comment by sayboras** [comment]
> Were there any conflicts?

Yup :(, I need to drop a couple of changes as those files are not available in 1.12 :(

**Comment by sayboras** [comment]
/test-1.20-4.9

**Comment by sayboras** [comment]
runtime test failed https://jenkins.cilium.io/job/Cilium-PR-Runtime-net-next/5091/ due to the below test cases. I can't find any existing issue, re-run to confirm if it's flake.

```
	 [10:45:22] START: route_linux_test.go:234: RouteSuitePrivileged.TestListRules
	 [10:45:22] Inside new network ns
	 [10:45:22] route_linux_test.go:235:
	 [10:45:22]     testListRules4(c)
	 [10:45:22] route_linux_test.go:487:
	 [10:45:22]     c.Errorf("expected len: %d, got: %d\n%s\n", len(wantRules), len(rules), diff)
	 [10:45:22] ... Error: expected len: 3, got: 5
	 [10:45:22]   []netlink.Rule{
	 [10:45:22]   	{
	 [10:45:22] - 		Priority: -1,
	 [10:45:22] + 		Priority: 9,
	 [10:45:22]   		Family:   0,
	 [10:45:22] - 		Table:    255,
	 [10:45:22] + 		Table:    2004,
	 [10:45:22] - 		Mark:     -1,
	 [10:45:22] + 		Mark:     512,
	 [10:45:22] - 		Mask:     -1,
	 [10:45:22] + 		Mask:     3840,
	 [10:45:22]   		Tos:      0,
	 [10:45:22]   		TunID:    0,
	 [10:45:22]   		... // 12 identical fields
	 [10:45:22]   	},
	 [10:45:22] + 	s"ip rule 10: from all to all table 2005",
	 [10:45:22] + 	s"ip rule 100: from all to all table 255",
	 [10:45:22]   	{Priority: 32766, Table: 254, Mark: -1, Mask: -1, ...},
	 [10:45:22]   	{Priority: 32767, Table: 253, Mark: -1, Mask: -1, ...},
	 [10:45:22]   }
	 [10:45:22] 
	 [10:45:22] 
	 [10:45:22] 
	 [10:45:22] Closed new network ns
	 [10:45:22] Set back to previous network ns
	 [10:45:22] Inside new network ns
	 [10:45:22] Closed new network ns
	 [10:45:22] Set back to previous network ns
	 [10:45:22] FAIL: route_linux_test.go:234: RouteSuitePrivileged.TestListRules
```

**Comment by sayboras** [comment]
The above flake is tracked in https://github.com/cilium/cilium/issues/23869

File: operator/pkg/ingress/envoy_config.go [hunk 1]
```diff
@@ -627,8 +637,8 @@ func getRedirectRouteConfigurationResource(ingress *slim_networkingv1.Ingress) (
 // Currently, only TPC keep-alive related options are specified.
 //
 // Related references:
-//  - https://man7.org/linux/man-pages/man7/tcp.7.html
-//  - https://github.com/envoyproxy/envoy/issues/3634
+//   - https://man7.org/linux/man-pages/man7/tcp.7.html
+//   - https://github.com/envoyproxy/envoy/issues/3634
```
Comment (pchaigno) [inline operator/pkg/ingress/envoy_config.go:641]: What happened here? :thinking:

**Review by pchaigno — APPROVED** [review]

File: operator/pkg/ingress/envoy_config.go (hunk 1 above)
Comment (sayboras) [reply operator/pkg/ingress/envoy_config.go:641]: probably it's auto format that I have configured in IDE. Golang recently added format for godocs as well :(

**Pull request merged**